be/src/agent/agent_server.cpp

```diff
@@ -110,25 +116,43 @@ void AgentServer::Impl::init_or_die() {
         }
     }
 
-    auto st =
-            ThreadPoolBuilder("publish_version")
-                    .set_min_threads(config::transaction_publish_version_worker_count)
-                    .set_max_threads(config::transaction_publish_version_worker_count)
-                    // The ideal queue size of threadpool should be larger than the maximum number of tablet of a partition.
-                    // But it seems that there's no limit for the number of tablets of a partition.
-                    // Since a large queue size brings a little overhead, a big one is chosen here.
-                    .set_max_queue_size(2048)
-                    .build(&_thread_pool_publish_version);
-    CHECK(st.ok()) << st;
+#define BUILD_DYNAMIC_TASK_THREAD_POOL(name, min_threads, max_threads, queue_size, pool) \
+    do {                                                                                 \
+        auto st = ThreadPoolBuilder(name)                                                \
+                          .set_min_threads(min_threads)                                  \
+                          .set_max_threads(max_threads)                                  \
+                          .set_max_queue_size(queue_size)                                \
+                          .build(&(pool));                                               \
+        CHECK(st.ok()) << st;                                                            \
+    } while (false)
+
+    // The ideal queue size of threadpool should be larger than the maximum number of tablet of a partition.
+    // But it seems that there's no limit for the number of tablets of a partition.
+    // Since a large queue size brings a little overhead, a big one is chosen here.
+    BUILD_DYNAMIC_TASK_THREAD_POOL("publish_version", config::transaction_publish_version_worker_count,
+                                   config::transaction_publish_version_worker_count,
+                                   DEFAULT_DYNAMIC_THREAD_POOL_QUEUE_SIZE, _thread_pool_publish_version);
+#ifndef BE_TEST
```

Comment:
this macro definition is useless.

Reply:
it's necessary, because currently the corresponding task work pool, e.g. `_clone_workers` will only be created if it's not a test build.